miner/worker.go

```diff
@@ -915,6 +958,21 @@ func (w *worker) commitTransactions(txs *types.TransactionsByPriceAndNonce, coin
 			log.Trace("Skipping unsupported transaction type", "sender", from, "type", tx.Type())
 			txs.Pop()
 
+		case errors.Is(err, circuitcapacitychecker.ErrBlockRowConsumptionOverflow):
+			// Circuit capacity limit reached in a block, don't pop or shift, just quit the loop immediately
+			log.Trace("Circuit capacity limit reached in a block", "acc_rows", w.current.accRows, "tx", tx.Hash())
+			break loop
+
+		case errors.Is(err, circuitcapacitychecker.ErrTxRowConsumptionOverflow):
```

Comment:
my attempt to fix this: https://github.com/scroll-tech/go-ethereum/pull/412

do you think it can address the concern?

Reply:
It seems this won't work if we restart l2geth. I think we might need to persist this info in DB.

Reply:
it won't be the accurate "max" if we restart l2geth, but it will eventually become accurate.

let's say at the start `maxSkippedL1MsgIndex` is `100`.
if no future tx will be skipped, we are all good.
if there is a L1MsgTx with index=105 being skipped, `maxSkippedL1MsgIndex` will be marked as 105.

doing it in memory instead of persistence can save some overheads.

Reply:
Isn't it more like this?
1. Last included index is `100`
2. We skip `100` messages, `maxSkippedL1MsgIndex` is `200`
3. Node crashes, restarts
4. Now `worker` starts again from `QueueIndex=101`, will not include any messages for 5 more blocks (assuming `NumL1MessagesPerBlock=20`)
Maybe it's not a huge issue that we need to solve in this PR.

Reply:
make sense. I will fix it.

Reply:
updated in https://github.com/scroll-tech/go-ethereum/pull/412